apps/dashboard/src/components/workflow-row.tsx

```diff
@@ -0,0 +1,239 @@
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button variant="ghost" size="icon">
+              <RiMore2Fill />
+            </Button>
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="w-56">
+            <DropdownMenuGroup>
+              <DropdownMenuItem>
+                <RiPlayCircleLine />
+                Trigger workflow
+              </DropdownMenuItem>
+              <PromoteWorkflowMenuItem
+                currentEnvironment={currentEnvironment}
+                isPromotable={isPromotable}
+                tooltipContent={tooltipContent}
+                onPromote={handlePromote}
+              />
+              <DropdownMenuItem>
+                <RiPulseFill />
+                View activity
+              </DropdownMenuItem>
+            </DropdownMenuGroup>
+            <DropdownMenuSeparator />
+            <DropdownMenuGroup>
+              <DropdownMenuItem>
+                <RiPauseCircleLine />
+                Pause workflow
+              </DropdownMenuItem>
+              <DropdownMenuItem className="text-destructive">
+                <RiDeleteBin2Line />
+                Delete workflow
+              </DropdownMenuItem>
+            </DropdownMenuGroup>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+const PromoteWorkflowMenuItem = ({
+  currentEnvironment,
+  isPromotable,
+  tooltipContent,
+  onPromote,
+}: {
+  currentEnvironment: IEnvironment | undefined;
+  isPromotable: boolean;
+  tooltipContent: string | undefined;
+  onPromote: () => void;
+}) => {
+  if (!currentEnvironment || currentEnvironment.name === 'Production') {
+    return null;
+  }
+
+  if (isPromotable) {
+    return (
+      <DropdownMenuItem onClick={onPromote}>
+        <RiGitPullRequestFill />
+        Promote to Production
+      </DropdownMenuItem>
+    );
+  }
+
+  return (
+    <TooltipProvider delayDuration={100}>
+      <Tooltip>
+        <TooltipTrigger>
+          <DropdownMenuItem disabled>
+            <RiGitPullRequestFill />
+            Promote to Production
+          </DropdownMenuItem>
+        </TooltipTrigger>
+        <TooltipPortal>
+          <TooltipContent>{tooltipContent}</TooltipContent>
+        </TooltipPortal>
+      </Tooltip>
+    </TooltipProvider>
+  );
+};
+
+const PromoteWorkflowConfirmModal = ({
+  open,
+  setOpen,
+  promoteWorkflow,
+}: {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  promoteWorkflow: UseMutateAsyncFunction<WorkflowResponseDto, unknown, void, void>;
+}) => {
+  async function onConfirm() {
+    setOpen(false);
+    await promoteWorkflow();
+  }
+
+  return (
+    <Dialog modal open={open}>
+      <DialogPortal>
+        <DialogOverlay />
+        <DialogContent className="sm:max-w-[440px]">
+          <div className="flex items-start gap-4 self-stretch">
+            <div className="flex items-center justify-center gap-2 rounded-[10px] bg-[#FF84471A] p-2">
```

Comment:
please do not use the hardcoded values for colors, `bg-warning/10` should work in this case